InterruptingEventSubprocessTest.shouldCloseEventSubscriptions flaky

**Summary**

I run the tests via mvn and the parallel profile.

> Try to answer the following as best as possible

- How often does the test fail? Once, locally
- Does it block your work? idk
- Do we suspect that it is a real failure? idk

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
Paste the failure here
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

<details><summary>Logs</summary>
<pre>

[INFO] Running io.zeebe.engine.processing.bpmn.subprocess.InterruptingEventSubprocessTest
[ERROR] Tests run: 21, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.69 s <<< FAILURE! - in io.zeebe.engine.processing.bpmn.subprocess.InterruptingEventSubprocessTest
[ERROR] io.zeebe.engine.processing.bpmn.subprocess.InterruptingEventSubprocessTest.shouldCloseEventSubscriptions[timer event subprocess]  Time elapsed: 0.556 s  <<< FAILURE!
java.lang.AssertionError: 

Expecting ArrayList:
 <[OPEN, OPEN, OPENED, OPEN]>
to contain:
 <[CLOSED]>
but could not find the following element(s):
 <[CLOSED]>

	at io.zeebe.engine.processing.bpmn.subprocess.InterruptingEventSubprocessTest.shouldCloseEventSubscriptions(InterruptingEventSubprocessTest.java:346)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:157)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)

</pre>
</details>

[io.zeebe.engine.processing.bpmn.subprocess.InterruptingEventSubprocessTest.txt](https://github.com/zeebe-io/zeebe/files/4936983/io.zeebe.engine.processing.bpmn.subprocess.InterruptingEventSubprocessTest.txt)
[io.zeebe.engine.processing.bpmn.subprocess.InterruptingEventSubprocessTest-output.txt](https://github.com/zeebe-io/zeebe/files/4936984/io.zeebe.engine.processing.bpmn.subprocess.InterruptingEventSubprocessTest-output.txt)

```
11:18:12.628 [] INFO  io.zeebe.test - Test failed, following records were exported:
11:18:12.735 [] INFO  io.zeebe.test - {"valueType":"DEPLOYMENT","key":-1,"position":1,"timestamp":1595236692585,"recordType":"COMMAND","intent":"CREATE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"resources":[{"resource":"PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxkZWZpbml0aW9ucyB4bWxuczpicG1uZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvREkiIHhtbG5zOmRjPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9ERC8yMDEwMDUyNC9EQyIgeG1sbnM6ZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0RELzIwMTAwNTI0L0RJIiB4bWxuczpuczA9Imh0dHA6Ly9jYW11bmRhLm9yZy9zY2hlbWEvemVlYmUvMS4wIiBpZD0iZGVmaW5pdGlvbnNfMjdkODVjMTgtNjk2NC00YmM5LThiZjEtZTk1MjJhZTdhNzU4IiB0YXJnZXROYW1lc3BhY2U9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvTU9ERUwiIHhtbG5zPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9CUE1OLzIwMTAwNTI0L01PREVMIj4KICA8cHJvY2VzcyBpZD0icHJvYyIgaXNFeGVjdXRhYmxlPSJ0cnVlIj4KICAgIDxzdWJQcm9jZXNzIGlkPSJldmVudF9zdWJfcHJvYyIgbmFtZT0iZXZlbnRfc3ViX3Byb2MiIHRyaWdnZXJlZEJ5RXZlbnQ9InRydWUiPgogICAgICA8c3RhcnRFdmVudCBpZD0iZXZlbnRfc3ViX3N0YXJ0IiBpc0...
11:18:12.737 [] INFO  io.zeebe.test - {"valueType":"DEPLOYMENT","key":2251799813685250,"position":2,"timestamp":1595246232598,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"resources":[{"resource":"PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxkZWZpbml0aW9ucyB4bWxuczpicG1uZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvREkiIHhtbG5zOmRjPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9ERC8yMDEwMDUyNC9EQyIgeG1sbnM6ZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0RELzIwMTAwNTI0L0RJIiB4bWxuczpuczA9Imh0dHA6Ly9jYW11bmRhLm9yZy9zY2hlbWEvemVlYmUvMS4wIiBpZD0iZGVmaW5pdGlvbnNfMjdkODVjMTgtNjk2NC00YmM5LThiZjEtZTk1MjJhZTdhNzU4IiB0YXJnZXROYW1lc3BhY2U9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvTU9ERUwiIHhtbG5zPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9CUE1OLzIwMTAwNTI0L01PREVMIj4KICA8cHJvY2VzcyBpZD0icHJvYyIgaXNFeGVjdXRhYmxlPSJ0cnVlIj4KICAgIDxzdWJQcm9jZXNzIGlkPSJldmVudF9zdWJfcHJvYyIgbmFtZT0iZXZlbnRfc3ViX3Byb2MiIHRyaWdnZXJlZEJ5RXZlbnQ9InRydWUiPgogICAgICA8c3RhcnRFdmVudCBpZD0iZXZlbnRfc3ViX...
11:18:12.737 [] INFO  io.zeebe.test - {"valueType":"DEPLOYMENT","key":2251799813685250,"position":3,"timestamp":1595246232598,"recordType":"COMMAND","intent":"DISTRIBUTE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"resources":[{"resource":"PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxkZWZpbml0aW9ucyB4bWxuczpicG1uZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvREkiIHhtbG5zOmRjPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9ERC8yMDEwMDUyNC9EQyIgeG1sbnM6ZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0RELzIwMTAwNTI0L0RJIiB4bWxuczpuczA9Imh0dHA6Ly9jYW11bmRhLm9yZy9zY2hlbWEvemVlYmUvMS4wIiBpZD0iZGVmaW5pdGlvbnNfMjdkODVjMTgtNjk2NC00YmM5LThiZjEtZTk1MjJhZTdhNzU4IiB0YXJnZXROYW1lc3BhY2U9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvTU9ERUwiIHhtbG5zPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9CUE1OLzIwMTAwNTI0L01PREVMIj4KICA8cHJvY2VzcyBpZD0icHJvYyIgaXNFeGVjdXRhYmxlPSJ0cnVlIj4KICAgIDxzdWJQcm9jZXNzIGlkPSJldmVudF9zdWJfcHJvYyIgbmFtZT0iZXZlbnRfc3ViX3Byb2MiIHRyaWdnZXJlZEJ5RXZlbnQ9InRydWUiPgogICAgICA8c3RhcnRFdmVudCBpZD0iZXZlbnRf...
11:18:12.738 [] INFO  io.zeebe.test - {"valueType":"DEPLOYMENT","key":2251799813685250,"position":4,"timestamp":1595246232598,"recordType":"EVENT","intent":"DISTRIBUTED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"resources":[{"resource":"PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjxkZWZpbml0aW9ucyB4bWxuczpicG1uZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvREkiIHhtbG5zOmRjPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9ERC8yMDEwMDUyNC9EQyIgeG1sbnM6ZGk9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0RELzIwMTAwNTI0L0RJIiB4bWxuczpuczA9Imh0dHA6Ly9jYW11bmRhLm9yZy9zY2hlbWEvemVlYmUvMS4wIiBpZD0iZGVmaW5pdGlvbnNfMjdkODVjMTgtNjk2NC00YmM5LThiZjEtZTk1MjJhZTdhNzU4IiB0YXJnZXROYW1lc3BhY2U9Imh0dHA6Ly93d3cub21nLm9yZy9zcGVjL0JQTU4vMjAxMDA1MjQvTU9ERUwiIHhtbG5zPSJodHRwOi8vd3d3Lm9tZy5vcmcvc3BlYy9CUE1OLzIwMTAwNTI0L01PREVMIj4KICA8cHJvY2VzcyBpZD0icHJvYyIgaXNFeGVjdXRhYmxlPSJ0cnVlIj4KICAgIDxzdWJQcm9jZXNzIGlkPSJldmVudF9zdWJfcHJvYyIgbmFtZT0iZXZlbnRfc3ViX3Byb2MiIHRyaWdnZXJlZEJ5RXZlbnQ9InRydWUiPgogICAgICA8c3RhcnRFdmVudCBpZD0iZXZlbnRfc...
11:18:12.765 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE_CREATION","key":-1,"position":5,"timestamp":1595236692599,"recordType":"COMMAND","intent":"CREATE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":-1,"workflowKey":-1,"variables":{"key":123},"workflowInstanceKey":-1,"bpmnProcessId":"proc"},"sourceRecordPosition":-1}
11:18:12.768 [] INFO  io.zeebe.test - {"valueType":"VARIABLE","key":2251799813685252,"position":6,"timestamp":1595246232599,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"name":"key","value":"123","workflowKey":2251799813685249,"workflowInstanceKey":2251799813685251,"scopeKey":2251799813685251},"sourceRecordPosition":5}
11:18:12.771 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685251,"position":7,"timestamp":1595246232599,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":-1,"elementId":"proc","bpmnElementType":"PROCESS","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":5}
11:18:12.771 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE_CREATION","key":2251799813685253,"position":8,"timestamp":1595246232599,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"variables":{"key":123},"workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc"},"sourceRecordPosition":5}
11:18:12.773 [] INFO  io.zeebe.test - {"valueType":"TIMER","key":-1,"position":9,"timestamp":1595246232599,"recordType":"COMMAND","intent":"CREATE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"targetElementId":"event_sub_start","workflowKey":2251799813685249,"workflowInstanceKey":2251799813685251,"elementInstanceKey":2251799813685251,"dueDate":1595246292599,"repetitions":1},"sourceRecordPosition":7}
11:18:12.773 [] INFO  io.zeebe.test - {"valueType":"TIMER","key":-1,"position":10,"timestamp":1595246232599,"recordType":"COMMAND","intent":"CREATE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"targetElementId":"other-timer","workflowKey":2251799813685249,"workflowInstanceKey":2251799813685251,"elementInstanceKey":2251799813685251,"dueDate":1595332632599,"repetitions":1},"sourceRecordPosition":7}
11:18:12.773 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685251,"position":11,"timestamp":1595246232599,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":-1,"elementId":"proc","bpmnElementType":"PROCESS","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":7}
11:18:12.773 [] INFO  io.zeebe.test - {"valueType":"TIMER","key":2251799813685254,"position":12,"timestamp":1595246232600,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"targetElementId":"event_sub_start","workflowKey":2251799813685249,"workflowInstanceKey":2251799813685251,"elementInstanceKey":2251799813685251,"dueDate":1595246292599,"repetitions":1},"sourceRecordPosition":9}
11:18:12.773 [] INFO  io.zeebe.test - {"valueType":"TIMER","key":2251799813685255,"position":13,"timestamp":1595246232600,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"targetElementId":"other-timer","workflowKey":2251799813685249,"workflowInstanceKey":2251799813685251,"elementInstanceKey":2251799813685251,"dueDate":1595332632599,"repetitions":1},"sourceRecordPosition":10}
11:18:12.774 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685256,"position":14,"timestamp":1595246232601,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"start_proc","bpmnElementType":"START_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":11}
11:18:12.774 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685256,"position":15,"timestamp":1595246232613,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"start_proc","bpmnElementType":"START_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":14}
11:18:12.776 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685256,"position":16,"timestamp":1595246232614,"recordType":"EVENT","intent":"ELEMENT_COMPLETING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"start_proc","bpmnElementType":"START_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":15}
11:18:12.776 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685256,"position":17,"timestamp":1595246232614,"recordType":"EVENT","intent":"ELEMENT_COMPLETED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"start_proc","bpmnElementType":"START_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":16}
11:18:12.776 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685257,"position":18,"timestamp":1595246232614,"recordType":"EVENT","intent":"SEQUENCE_FLOW_TAKEN","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"sequenceFlow_00aa177d-abed-4587-b93a-9ed0a0e83ae7","bpmnElementType":"SEQUENCE_FLOW","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":17}
11:18:12.776 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685258,"position":19,"timestamp":1595246232614,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"task","bpmnElementType":"SERVICE_TASK","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":18}
11:18:12.777 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685258,"position":20,"timestamp":1595246232615,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"task","bpmnElementType":"SERVICE_TASK","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":19}
11:18:12.780 [] INFO  io.zeebe.test - {"valueType":"JOB","key":-1,"position":21,"timestamp":1595246232615,"recordType":"COMMAND","intent":"CREATE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"errorMessage":"","type":"type","workflowKey":2251799813685249,"variables":{},"elementId":"task","retries":3,"errorCode":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","deadline":-1,"elementInstanceKey":2251799813685258,"worker":"","customHeaders":{},"workflowDefinitionVersion":1},"sourceRecordPosition":20}
11:18:12.780 [] INFO  io.zeebe.test - {"valueType":"JOB","key":2251799813685259,"position":22,"timestamp":1595246232615,"recordType":"EVENT","intent":"CREATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"errorMessage":"","type":"type","workflowKey":2251799813685249,"variables":{},"elementId":"task","retries":3,"errorCode":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","deadline":-1,"elementInstanceKey":2251799813685258,"worker":"","customHeaders":{},"workflowDefinitionVersion":1},"sourceRecordPosition":21}
11:18:12.780 [] INFO  io.zeebe.test - {"valueType":"TIMER","key":2251799813685254,"position":23,"timestamp":1595246292615,"recordType":"COMMAND","intent":"TRIGGER","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"targetElementId":"event_sub_start","workflowKey":2251799813685249,"workflowInstanceKey":2251799813685251,"elementInstanceKey":2251799813685251,"dueDate":1595246292599,"repetitions":1},"sourceRecordPosition":22}
11:18:12.780 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685261,"position":24,"timestamp":1595246292616,"recordType":"EVENT","intent":"EVENT_OCCURRED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"event_sub_start","bpmnElementType":"START_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":23}
11:18:12.780 [] INFO  io.zeebe.test - {"valueType":"TIMER","key":2251799813685254,"position":25,"timestamp":1595246292616,"recordType":"EVENT","intent":"TRIGGERED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"targetElementId":"event_sub_start","workflowKey":2251799813685249,"workflowInstanceKey":2251799813685251,"elementInstanceKey":2251799813685251,"dueDate":1595246292599,"repetitions":1},"sourceRecordPosition":23}
11:18:12.782 [] INFO  io.zeebe.test - {"valueType":"MESSAGE_SUBSCRIPTION","key":-1,"position":26,"timestamp":1595236692617,"recordType":"COMMAND","intent":"OPEN","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"messageName":"other-message","correlationKey":"123","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":-1}
11:18:12.782 [] INFO  io.zeebe.test - {"valueType":"MESSAGE_SUBSCRIPTION","key":-1,"position":27,"timestamp":1595236692617,"recordType":"COMMAND","intent":"OPEN","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"messageName":"other-message","correlationKey":"123","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":-1}
11:18:12.782 [] INFO  io.zeebe.test - {"valueType":"TIMER","key":2251799813685255,"position":28,"timestamp":1595246292618,"recordType":"COMMAND","intent":"CANCEL","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"targetElementId":"other-timer","workflowKey":2251799813685249,"workflowInstanceKey":2251799813685251,"elementInstanceKey":2251799813685251,"dueDate":1595332632599,"repetitions":1},"sourceRecordPosition":24}
11:18:12.783 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685258,"position":29,"timestamp":1595246292618,"recordType":"EVENT","intent":"ELEMENT_TERMINATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"task","bpmnElementType":"SERVICE_TASK","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":24}
11:18:12.783 [] INFO  io.zeebe.test - {"valueType":"MESSAGE_SUBSCRIPTION","key":-1,"position":30,"timestamp":1595236692618,"recordType":"COMMAND","intent":"CLOSE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"messageName":"other-message","correlationKey":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":-1}
11:18:12.783 [] INFO  io.zeebe.test - {"valueType":"MESSAGE_SUBSCRIPTION","key":-1,"position":31,"timestamp":1595246292619,"recordType":"EVENT","intent":"OPENED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"messageName":"other-message","correlationKey":"123","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":26}
11:18:12.785 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE_SUBSCRIPTION","key":-1,"position":32,"timestamp":1595236692619,"recordType":"COMMAND","intent":"OPEN","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"messageName":"other-message","variables":{},"correlationKey":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":-1}
11:18:12.785 [] INFO  io.zeebe.test - {"valueType":"MESSAGE_SUBSCRIPTION","key":-1,"position":33,"timestamp":1595246292619,"recordType":"COMMAND_REJECTION","intent":"OPEN","partitionId":1,"rejectionType":"INVALID_STATE","rejectionReason":"Expected to open a new message subscription for element with key '2251799813685251' and message name 'other-message', but there is already a message subscription for that element key and message name opened","value":{"messageName":"other-message","correlationKey":"123","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":27}
11:18:12.785 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE_SUBSCRIPTION","key":-1,"position":34,"timestamp":1595236692619,"recordType":"COMMAND","intent":"OPEN","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"messageName":"other-message","variables":{},"correlationKey":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":-1}
11:18:12.785 [] INFO  io.zeebe.test - {"valueType":"TIMER","key":2251799813685255,"position":35,"timestamp":1595246292619,"recordType":"EVENT","intent":"CANCELED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"targetElementId":"other-timer","workflowKey":2251799813685249,"workflowInstanceKey":2251799813685251,"elementInstanceKey":2251799813685251,"dueDate":1595332632599,"repetitions":1},"sourceRecordPosition":28}
11:18:12.786 [] INFO  io.zeebe.test - {"valueType":"JOB","key":2251799813685259,"position":36,"timestamp":1595246292620,"recordType":"COMMAND","intent":"CANCEL","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"errorMessage":"","type":"type","workflowKey":2251799813685249,"variables":{},"elementId":"task","retries":3,"errorCode":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","deadline":-1,"elementInstanceKey":2251799813685258,"worker":"","customHeaders":{},"workflowDefinitionVersion":1},"sourceRecordPosition":29}
11:18:12.786 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685258,"position":37,"timestamp":1595246292620,"recordType":"EVENT","intent":"ELEMENT_TERMINATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"task","bpmnElementType":"SERVICE_TASK","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":29}
11:18:12.786 [] INFO  io.zeebe.test - {"valueType":"MESSAGE_SUBSCRIPTION","key":-1,"position":38,"timestamp":1595246292620,"recordType":"EVENT","intent":"CLOSED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"messageName":"other-message","correlationKey":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":30}
11:18:12.786 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE_SUBSCRIPTION","key":-1,"position":39,"timestamp":1595236692620,"recordType":"COMMAND","intent":"CLOSE","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"messageName":"other-message","variables":{},"correlationKey":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":-1}
11:18:12.786 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE_SUBSCRIPTION","key":-1,"position":40,"timestamp":1595246292621,"recordType":"COMMAND_REJECTION","intent":"OPEN","partitionId":1,"rejectionType":"INVALID_STATE","rejectionReason":"Expected to open workflow instance subscription with element key '2251799813685251' and message name 'other-message', but it is already closing","value":{"messageName":"other-message","variables":{},"correlationKey":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":32}
11:18:12.787 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE_SUBSCRIPTION","key":-1,"position":41,"timestamp":1595246292621,"recordType":"COMMAND_REJECTION","intent":"OPEN","partitionId":1,"rejectionType":"INVALID_STATE","rejectionReason":"Expected to open workflow instance subscription with element key '2251799813685251' and message name 'other-message', but it is already closing","value":{"messageName":"other-message","variables":{},"correlationKey":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":34}
11:18:12.787 [] INFO  io.zeebe.test - {"valueType":"JOB","key":2251799813685259,"position":42,"timestamp":1595246292621,"recordType":"EVENT","intent":"CANCELED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"errorMessage":"","type":"type","workflowKey":2251799813685249,"variables":{},"elementId":"task","retries":3,"errorCode":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","deadline":-1,"elementInstanceKey":2251799813685258,"worker":"","customHeaders":{},"workflowDefinitionVersion":1},"sourceRecordPosition":36}
11:18:12.787 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685262,"position":43,"timestamp":1595246292622,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"event_sub_proc","bpmnElementType":"SUB_PROCESS","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":37}
11:18:12.787 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE_SUBSCRIPTION","key":-1,"position":44,"timestamp":1595246292622,"recordType":"EVENT","intent":"CLOSED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"messageName":"other-message","variables":{},"correlationKey":"","workflowInstanceKey":2251799813685251,"bpmnProcessId":"","elementInstanceKey":2251799813685251,"messageKey":-1},"sourceRecordPosition":39}
11:18:12.787 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685262,"position":45,"timestamp":1595246292622,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"event_sub_proc","bpmnElementType":"SUB_PROCESS","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":43}
11:18:12.787 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685263,"position":46,"timestamp":1595246292623,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685262,"elementId":"event_sub_start","bpmnElementType":"START_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":45}
11:18:12.789 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685263,"position":47,"timestamp":1595246292623,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685262,"elementId":"event_sub_start","bpmnElementType":"START_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":46}
11:18:12.789 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685263,"position":48,"timestamp":1595246292623,"recordType":"EVENT","intent":"ELEMENT_COMPLETING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685262,"elementId":"event_sub_start","bpmnElementType":"START_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":47}
11:18:12.789 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685263,"position":49,"timestamp":1595246292623,"recordType":"EVENT","intent":"ELEMENT_COMPLETED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685262,"elementId":"event_sub_start","bpmnElementType":"START_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":48}
11:18:12.789 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685264,"position":50,"timestamp":1595246292624,"recordType":"EVENT","intent":"SEQUENCE_FLOW_TAKEN","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685262,"elementId":"sequenceFlow_11cfe0cd-79c1-478c-8fd6-58a8e11e403c","bpmnElementType":"SEQUENCE_FLOW","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":49}
11:18:12.790 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685265,"position":51,"timestamp":1595246292624,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685262,"elementId":"event_sub_end","bpmnElementType":"END_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":50}
11:18:12.790 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685265,"position":52,"timestamp":1595246292624,"recordType":"EVENT","intent":"ELEMENT_ACTIVATED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685262,"elementId":"event_sub_end","bpmnElementType":"END_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":51}
11:18:12.790 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685265,"position":53,"timestamp":1595246292625,"recordType":"EVENT","intent":"ELEMENT_COMPLETING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685262,"elementId":"event_sub_end","bpmnElementType":"END_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":52}
11:18:12.790 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685265,"position":54,"timestamp":1595246292625,"recordType":"EVENT","intent":"ELEMENT_COMPLETED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685262,"elementId":"event_sub_end","bpmnElementType":"END_EVENT","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":53}
11:18:12.790 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685262,"position":55,"timestamp":1595246292625,"recordType":"EVENT","intent":"ELEMENT_COMPLETING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"event_sub_proc","bpmnElementType":"SUB_PROCESS","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":54}
11:18:12.790 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685262,"position":56,"timestamp":1595246292625,"recordType":"EVENT","intent":"ELEMENT_COMPLETED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":2251799813685251,"elementId":"event_sub_proc","bpmnElementType":"SUB_PROCESS","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":55}
11:18:12.790 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685251,"position":57,"timestamp":1595246292626,"recordType":"EVENT","intent":"ELEMENT_COMPLETING","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":-1,"elementId":"proc","bpmnElementType":"PROCESS","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":56}
11:18:12.790 [] INFO  io.zeebe.test - {"valueType":"WORKFLOW_INSTANCE","key":2251799813685251,"position":58,"timestamp":1595246292626,"recordType":"EVENT","intent":"ELEMENT_COMPLETED","partitionId":1,"rejectionType":"NULL_VAL","rejectionReason":"","value":{"version":1,"workflowKey":2251799813685249,"flowScopeKey":-1,"elementId":"proc","bpmnElementType":"PROCESS","workflowInstanceKey":2251799813685251,"bpmnProcessId":"proc","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1},"sourceRecordPosition":57}

java.lang.AssertionError: 
Expecting ArrayList:
 <[OPEN, OPEN, CLOSE, OPENED]>
to contain:
 <[CLOSED]>
but could not find the following element(s):
 <[CLOSED]>


	at io.zeebe.engine.processing.bpmn.subprocess.InterruptingEventSubprocessTest.shouldCloseEventSubscriptions(InterruptingEventSubprocessTest.java:346)
```

--- a/engine/src/test/java/io/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessTest.java
+++ b/engine/src/test/java/io/zeebe/engine/processing/bpmn/subprocess/InterruptingEventSubprocessTest.java
@@ -334,22 +334,31 @@ public class InterruptingEventSubprocessTest {
             "timer-event-subprocess",
             s -> s.startEvent("other-timer").timerWithDuration("P1D").endEvent());
 
-    final long wfInstanceKey = createInstanceAndTriggerEvent(workflow(eventSubprocess));
+    final long wfInstanceKey = createInstanceAndWaitForTask(workflow(eventSubprocess));
+
+    RecordingExporter.messageSubscriptionRecords(MessageSubscriptionIntent.OPENED)
+        .withWorkflowInstanceKey(wfInstanceKey)
+        .withMessageName("other-message")
+        .await();
+
+    triggerEventSubprocess.accept(wfInstanceKey);
 
     // then
     assertThat(
-            RecordingExporter.messageSubscriptionRecords()
+            RecordingExporter.records()
+                .limitToWorkflowInstance(wfInstanceKey)
+                .messageSubscriptionRecords()
                 .withWorkflowInstanceKey(wfInstanceKey)
-                .withMessageName("other-message")
-                .limit(4))
+                .withMessageName("other-message"))
         .extracting(Record::getIntent)
         .contains(MessageSubscriptionIntent.CLOSED);
 
     assertThat(
-            RecordingExporter.timerRecords()
+            RecordingExporter.records()
+                .limitToWorkflowInstance(wfInstanceKey)
+                .timerRecords()
                 .withWorkflowInstanceKey(wfInstanceKey)
-                .withHandlerNodeId("other-timer")
-                .limit(4))
+                .withHandlerNodeId("other-timer"))
         .extracting(Record::getIntent)
         .contains(TimerIntent.CANCELED);
   }

--- a/engine/src/test/java/io/zeebe/engine/util/EngineRule.java
+++ b/engine/src/test/java/io/zeebe/engine/util/EngineRule.java
@@ -71,7 +71,7 @@ public final class EngineRule extends ExternalResource {
 
   private static final int PARTITION_ID = Protocol.DEPLOYMENT_PARTITION;
   private static final RecordingExporter RECORDING_EXPORTER = new RecordingExporter();
-  private StreamProcessorRule environmentRule;
+  private final StreamProcessorRule environmentRule;
   private final RecordingExporterTestWatcher recordingExporterTestWatcher =
       new RecordingExporterTestWatcher();
   private final int partitionCount;
@@ -80,7 +80,7 @@ public final class EngineRule extends ExternalResource {
 
   private final Int2ObjectHashMap<SubscriptionCommandMessageHandler> subscriptionHandlers =
       new Int2ObjectHashMap<>();
-  private final ExecutorService subscriptionHandlerExecutor = Executors.newSingleThreadExecutor();
+  private ExecutorService subscriptionHandlerExecutor;
 
   private EngineRule(final int partitionCount) {
     this(partitionCount, false);
@@ -115,6 +115,8 @@ public final class EngineRule extends ExternalResource {
 
   @Override
   protected void before() {
+    subscriptionHandlerExecutor = Executors.newSingleThreadExecutor();
+
     if (!explicitStart) {
       startProcessors();
     }
@@ -123,7 +125,6 @@ public final class EngineRule extends ExternalResource {
   @Override
   protected void after() {
     subscriptionHandlerExecutor.shutdown();
-    environmentRule = null;
     subscriptionHandlers.clear();
   }
 

--- a/engine/src/test/java/io/zeebe/engine/util/StreamProcessorRule.java
+++ b/engine/src/test/java/io/zeebe/engine/util/StreamProcessorRule.java
@@ -248,6 +248,7 @@ public final class StreamProcessorRule implements TestRule {
     @Override
     protected void after() {
       streams = null;
+      streamProcessingComposite = null;
     }
   }
 

--- a/test-util/src/main/java/io/zeebe/test/util/record/RecordStream.java
+++ b/test-util/src/main/java/io/zeebe/test/util/record/RecordStream.java
@@ -81,4 +81,9 @@ public final class RecordStream extends ExporterRecordStream<RecordValue, Record
     return new IncidentRecordStream(
         filter(r -> r.getValueType() == ValueType.INCIDENT).map(Record.class::cast));
   }
+
+  public MessageSubscriptionRecordStream messageSubscriptionRecords() {
+    return new MessageSubscriptionRecordStream(
+        filter(r -> r.getValueType() == ValueType.MESSAGE_SUBSCRIPTION).map(Record.class::cast));
+  }
 }

